### PR TITLE
Port calls to API from Smooch Core

### PR DESCRIPTION
**Pull request by lemieux**
This PR removes the dependency to smooch-core and ports over all the api calls inside the SDK.

### CODE REVIEW

File: src/frame/js/actions/integrations.js [hunk 1]
```diff
@@ -229,17 +261,14 @@ export function fetchMessageBirdAttributes() {
...
-            .then(({appUser}) => {
-                dispatch(updateUser(appUser));
-
-                if (appUser.conversationStarted) {
-                    return dispatch(handleConversationUpdated());
-                }
-            })
-            .then(() => {
+        const {config: {appId}, user: {_id, pendingClients}} = getState();
+        return dispatch(http('POST', `/client/apps/${appId}/appusers/${_id}/clients`, data))
```
Comment (mspensieri) [inline src/frame/js/actions/integrations.js]: `/client` 👀 

Shows up a bunch of times in the rest of this file too

File: src/frame/js/actions/stripe.js [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
-import { core } from '../utils/core';
-import { getUserId } from './user';
+import http from './http';
 
 export function createTransaction(actionId, token) {
     return (dispatch, getState) => {
-        return core(getState()).appUsers.stripe.createTransaction(getUserId(getState()), actionId, token).catch((e) => {
-            dispatch(showErrorNotification(getState().ui.text.actionPaymentError));
+        const {config: {appId}, user: {_id}, ui} = getState();
+        return dispatch(http('GET', `/client/apps/${appId}/appusers/${_id}/stripe/transaction`, {
```
Comment (mspensieri) [inline src/frame/js/actions/stripe.js]: should be a `POST`

File: src/frame/js/actions/user.js [hunk 3]
```diff
@@ -46,13 +43,14 @@ export function immediateUpdate(props) {
...
-            return core(getState()).appUsers.update(getUserId(getState()), props).then((response) => {
-                dispatch(setUser(response.appUser));
-                if (updateToResolve) {
-                    updateToResolve(response);
-                }
-                return response;
-            });
+            return dispatch(http('PUT', `/apps/${appId}/appusers/${user._id}`, props))
+                .then((response) => {
+                    dispatch(setUser(response.appUser));
```
Comment (mspensieri) [inline src/frame/js/actions/user.js]: PUT user returns an empty body now

File: src/frame/js/actions/integrations.js [hunk 4]
```diff
@@ -250,11 +279,17 @@ export function linkSMSChannel(userId, data) {
...
     return (dispatch, getState) => {
-        return core(getState()).appUsers.unlinkChannel(userId, type)
+        const {config: {appId}, user: {_id, clients, pendingClients}} = getState();
+        const client = clients.find((client) => client.platform === type);
+
+        if (!client) {
+            return Promise.resolve();
+        }
+
+        return dispatch(http('DELETE', `/client/apps/${appId}/appusers/${_id}/clients/${client._id}`))
```
Comment (mspensieri) [inline src/frame/js/actions/integrations.js]: clients are accessed by their `.id` not `._id`. In fact I don't think we return `._id` in the API at all for a client. We might want to discuss this though

File: src/frame/js/actions/integrations.js (hunk 1 above)
Comment (lemieux) [inline src/frame/js/actions/integrations.js]: :man_facepalming:

File: src/frame/js/actions/integrations.js [hunk 5]
```diff
@@ -327,40 +364,24 @@ export function failSMSLink(error, type) {
     };
 }
 
-export function fetchViberQRCode() {
+export function fetchTransferRequestCode(type) {
     return (dispatch, getState) => {
-        const {integrations: {viber}} = getState();
+        const {user: {_id}, config: {integrations, appId}} = getState();
+        const integration = integrations.find(({type:_type}) => type === _type);
```
Comment (alavers) [inline src/frame/js/actions/integrations.js]: Deconstruction run amok! Isn't `.find((integration) => type === integration.type);` a fair bit clearer?

Comment (lemieux) [inline src/frame/js/actions/integrations.js]: yeah... I was thinking that too when I read it back lol

**Pull request merged**